pkg/component/runtime/service_command.go

```diff
@@ -93,11 +98,195 @@ func executeCommand(ctx context.Context, log *logger.Logger, binaryPath string,
 }
 
 func executeServiceCommand(ctx context.Context, log *logger.Logger, binaryPath string, spec *component.ServiceOperationsCommandSpec) error {
+	return executeServiceCommandWithRetries(
+		ctx, log, binaryPath, spec,
+		context.Background(), 20*time.Second, 15*time.Minute,
+	)
+}
+
+func executeServiceCommandWithRetries(
+	cmdCtx context.Context, log *logger.Logger, binaryPath string, spec *component.ServiceOperationsCommandSpec,
+	retryCtx context.Context, defaultRetrySleepInitDuration time.Duration, retrySleepMaxDuration time.Duration,
+) error {
 	if spec == nil {
 		log.Warnf("spec is nil, nothing to execute, binaryPath: %s", binaryPath)
 		return nil
 	}
-	return executeCommand(ctx, log, binaryPath, spec.Args, envSpecToEnv(spec.Env), spec.Timeout)
+
+	// If no initial sleep duration is specified, use default value
+	retrySleepInitDuration := spec.RetrySleepInitDuration
+	if retrySleepInitDuration == 0 {
+		retrySleepInitDuration = defaultRetrySleepInitDuration
+	}
+
+	serviceCmdRetrier.Start(
+		cmdCtx, log,
+		binaryPath, spec.Args, envSpecToEnv(spec.Env), spec.Timeout,
+		retryCtx, retrySleepInitDuration, retrySleepMaxDuration,
+	)
+	return nil
```

Comment:
no need to have error return if we're not returning error in any case

Reply:
Removed.